core/web/router.go

```diff
@@ -363,13 +363,16 @@ func uiCorsHandler(config orm.ConfigReader) gin.HandlerFunc {
 	}
 	if config.AllowOrigins() == "*" {
 		c.AllowAllOrigins = true
-	} else {
-		allowOrigins := strings.Split(config.AllowOrigins(), ",")
-		if len(allowOrigins) > 0 {
-			c.AllowOrigins = allowOrigins
+	} else if allowOrigins := strings.Split(config.AllowOrigins(), ","); len(allowOrigins) > 0 {
+		c.AllowOrigins = allowOrigins
+	}
+	handler := cors.New(c)
+	return func(c *gin.Context) {
+		handler(c)
+		if status := c.Writer.Status(); status >= 400 && status < 500 {
+			logger.Info("request denied by Cross-Origin Resource Sharing (CORS) policy")
```

Comment:
Isn't a CORS policy enforced by the browser?

Reply:
Good point